src/app/(nextjs_migration)/(authenticated)/user/settings/page.tsx

```diff
@@ -0,0 +1,249 @@
+};
+
+const deleteButton = (email: EmailRow) => {
+  const l10n = getL10n();
+
+  return (
+    <button
+      aria-label={l10n.getString("settings-delete-email-button")}
+      data-subscriber-id={email.subscriber_id}
+      data-email-id={email.id}
+      className="settings-email-remove-button js-remove-email-button"
+    >
+      <Image
+        src={ImageIconDelete}
+        alt={l10n.getString("settings-delete-email-button")}
+      />
+    </button>
+  );
+};
+
+const createEmailItem = (
+  email: EmailRow & { primary?: boolean },
+  breachCounts: Map<string, number>
+) => {
+  const l10n = getL10n();
+
+  return (
+    <li className="settings-email-item">
+      <strong>
+        {email.primary
+          ? l10n.getString("settings-email-label-primary", {
+              email: email.email,
+            })
+          : email.email}
+      </strong>
+      {email.verified
+        ? l10n.getString("settings-email-number-of-breaches-info", {
+            breachCount: breachCounts.get(email.email)!,
+          })
+        : emailNeedsVerificationSub(email)}
+      {email.primary ? null : deleteButton(email)}
+    </li>
+  );
+};
+
+// Moves the primary email to the front and sorts the rest alphabeticaly.
+const getSortedEmails = (emails: any[]) =>
+  [...emails].sort((a, b) => {
+    if (a.primary) {
+      return -1;
+    }
+
+    if (b.primary) {
+      return 1;
+    }
+
+    return a.email.localeCompare(b.email);
+  });
+
+const createEmailList = (
+  emails: EmailRow[],
+  breachCounts: Map<string, number>
+) => {
+  return (
+    <ul className="settings-email-list">
+      {getSortedEmails(emails).map((email) =>
+        createEmailItem(email, breachCounts)
+      )}
+    </ul>
+  );
+};
+
+const alertOptions = ({
+  allEmailsToPrimary,
+}: {
+  allEmailsToPrimary: boolean;
+}) => {
+  const l10n = getL10n();
+
+  return (
+    <div className="settings-alert-options">
+      <label className="settings-radio-input">
+        <input
+          defaultChecked={!allEmailsToPrimary}
+          className="js-settings-alert-options-input"
+          data-alert-option={0}
+          name="settings-alert-options"
+          type="radio"
+        />
+        <span className="settings-radio-label">
+          {l10n.getString("settings-alert-preferences-option-one")}
+        </span>
+      </label>
+
+      <label className="settings-radio-input">
+        <input
+          defaultChecked={allEmailsToPrimary}
+          className="js-settings-alert-options-input"
+          data-alert-option={1}
+          name="settings-alert-options"
+          type="radio"
+        />
+        <span className="settings-radio-label">
+          {l10n.getString("settings-alert-preferences-option-two")}
+        </span>
+      </label>
+    </div>
+  );
+};
+
+export default async function Settings() {
+  const l10n = getL10n();
+  const session = await getServerSession(authOptions);
+  if (!session || !session.user.subscriber) {
+    return redirect("/");
+  }
```

Comment:
Q: I'm a bit confused here.
I tried digging through the next-auth docs but wasn't sure what possible values `getServerSession()` returns. I think it something returns an empty object, the `!session.user.subscriber` call will throw errors about:
> _"TypeError: Cannot read properties of undefined (reading 'subscriber')"_

Reply:
Those docs aren't great, are they :/ But I think you're right, so I addd another guard for that: ef19f5fdb